ext/js/application.js

```diff
@@ -190,10 +190,35 @@ export class Application extends EventDispatcher {
      * @param {(application: Application) => (Promise<void>)} mainFunction
      */
     static async main(waitForDom, mainFunction) {
+        /** @type {MessagePort | null} */
+        // If this is Firefox, we don't have a service worker and can't postMessage,
+        // so we temporarily create a SharedWorker in order to establish a MessageChannel
+        // which we can use to postMessage with the backend.
+        // This can only be done in the extension context (aka iframe within popup),
+        // not in the content script context.
+        const backendPort = (!('serviceWorker' in navigator)) && window.location.protocol === new URL(import.meta.url).protocol ?
```

Comment:
That isn't quite what the check is doing. 
`!('serviceWorker' in navigator)` is kind of isFirefox, though I consider it to be a bit more broad as it's just checking the API that isn't available and in theory would support other browsers too. Ignoring that and naming this part isFirefox might be fine though since basically everything is either Firefox or Chromium-based.

`window.location.protocol === new URL(import.meta.url).protocol` on the other hand is checking that the code is executing within the iframe as opposed to the content script. `window.location.protocol` will be `chrome-extension` etc in this case which matches with `new URL(import.meta.url).protocol`.

I'll try to rename some stuff.